web/services/prometheus.go

```diff
@@ -36,11 +47,32 @@ func (p *prometheusService) GetHttpSDTargets() (models.PrometheusTargetsList, er
 		targets := &models.PrometheusTargets{
 			Targets: []string{fmt.Sprintf("%s:%d", host.SSHAddress, nodeExporterPort)},
 			Labels: map[string]string{
-				"hostname": host.Name,
+				"hostname":      host.Name,
+				"exporter_name": nodeExporterName,
 			},
 		}
 		targetsList = append(targetsList, targets)
 	}
 
 	return targetsList, nil
 }
+
+func (p *prometheusService) Query(query string, ts time.Time) (prometheusModel.Value, error) {
```

Comment:
This method aims to make the `Query` method easier. Otherwise, every time you want to query, you would need to have the context, check the warning, the errors, etc. We could make it a private function maybe.
It simply wants to reduce all of this duplicated code.

About the `GetExportersState`. Yes, we could move it to the prometheus service (in fact, I did it this way initially), but as I commented I changed this to put in the hosts service. Why? Because otherwise, I would need to have this function in the prometheus service, and later on create other function in the hosts service that only uses this function. I didn't want to pass the prometheus service to the hosts controller.  We could very well do that. It only adds other function (and I wanted to avoid this), but if you think that it makes the code better structured...